src/ripple/overlay/Message.h

```diff
@@ -48,14 +48,17 @@ namespace ripple {
 class Message : public std::enable_shared_from_this <Message>
 {
 public:
-    Message (::google::protobuf::Message const& message, int type);
+    Message (::google::protobuf::Message const& message, int type, bool compression_enabled);
 
 public:
     /** Retrieve the packed message data. */
     std::vector <uint8_t> const&
-    getBuffer () const
+    getBuffer (bool get_compressed) const
     {
-        return mBuffer;
+        if (get_compressed && mBufferCompressed.size() > 0)
+            return mBufferCompressed;
+        else
+            return mBuffer;
```

Comment:
I can specify `get_compressed` and still return an uncompressed buffer. We need to document that with a comment, and also rename the variable so it's clear it can return uncompressed data even when `compressed` is specified.

Reply:
Names are hard. I'm fine with `tryCompressed`.